BreadPlayer.Views.UWP/Core/SharedLogic.cs

```diff
@@ -47,7 +51,26 @@ public SharedLogic()
             InitializeCore.IsMobile = ApiInformation.IsApiContractPresent("Windows.Phone.PhoneContract", 1);
 
             InitializeCore.IsMobile = Window.Current?.Bounds.Width <= 600;
+
+            MediaDevice.DefaultAudioRenderDeviceChanged += OnDefaultAudioRenderDeviceChanged;
+        }
+
+        private async void OnDefaultAudioRenderDeviceChanged(object sender, DefaultAudioRenderDeviceChangedEventArgs args)
+        {
+            if (args.Role != AudioDeviceRole.Default || args.Id == _audioDeviceId)
+                return;
+
+            var oldDevice = await DeviceInformation.CreateFromIdAsync(_audioDeviceId);
+            var device = await DeviceInformation.CreateFromIdAsync(args.Id);
+
+            System.Diagnostics.Debug.WriteLine($"Switching audio render device from [{oldDevice.Name}] to [{device.Name}]");
+
+            _audioDeviceId = args.Id;
+
+            if (!InitializeCore.IsMobile)
```

Comment:
1. This `if statement` is redundant if you put an `if statement` around the event definition.
2. Move this whole event method to `ShellViewModel` just below where the `Shell_AudioEndpointChanged` event is.